### PR TITLE
bump cargo-toml to 0.15.2 and remove workarounds

### PR DESCRIPTION
**Pull request by skyzh**
Thanks to @kornelski for a quick review for my PR to cargo_toml and we can remove the workaround now!

### CODE REVIEW

File: src/manifest.rs [hunk 1]
```diff
@@ -8,33 +8,7 @@ pub(crate) struct Manifest {
...
-                *parsed
-                    .lib
-                    .as_mut()
-                    .expect("no lib element")
-                    .name
-                    .as_mut()
-                    .expect("no name in the lib") = name.clone();
-            }
-        }
+        let parsed = cargo_toml::Manifest::from_path(&path)?;
```
Comment (obi1kenobi) [inline src/manifest.rs:11]: Can you keep the original `.map_err()` call that adds a bit of context on the failing operation? Just pull the same error message that was here before on the parsing call before the workaround PR.

**Comment by skyzh** [comment]
Comment resolved! Actually we can use anyhow's `.context` here, which produces something like (as in [anyhow::Context](https://docs.rs/anyhow/latest/anyhow/trait.Context.html) example):

```
Error: Failed to read instrs from ./path/to/instrs.json

Caused by:
    No such file or directory (os error 2)
```

```rust
        let parsed = cargo_toml::Manifest::from_path(&path)
            .with_context(|e| format!("Failed when reading {}", path.display())?;
```

Do you feel this is a better way to throw an error? If so I can refactor map_err to context for manifest errors in this PR.

**Comment by obi1kenobi** [comment]
Yes, definitely! Thanks!

**Comment by skyzh** [comment]
done :)

**Review by obi1kenobi — APPROVED** [review]

**Comment by obi1kenobi** [comment]
Awesome! Thank you so much!

**Pull request merged**